packages/commonwealth/server/util/cosmosCache.ts

```diff
@@ -0,0 +1,103 @@
+import type { Request, RequestHandler } from 'express';
+import { cacheDecorator } from 'common-common/src/cacheDecorator';
+
+const oneBlock = 6; // typical Cosmos block time is ~6 seconds
+
+export const cosmosCacheRPC = (duration: number): RequestHandler => {
+  // RPC specific codes from cosmJS requests
+  const activeCodes = ['0801', '0802']; // ['DepositPerion', 'VotingPeriod']
+  const completedCodes = [
+    '0803', // 'Passed'
+    '0804', // 'Rejected'
+    '0805', // 'Failed'
+    '0803220a0a080000000000000087', // 'Passed - Osmosis'
+    '0803220a0a080000000000000100', // 'Passed - Osmosis'
+    '0803220a0a080000000000000189', // 'Passed - Osmosis'
+  ];
+
+  return async function cacheIfCacheworthyRPC(req, res, next) {
+    const body = parseReqBody(req);
+    const params = JSON.stringify(body?.params);
+    const method = body?.method;
+
+    if (/^(tx)/.test(method)) {
+      // TX Response: do not cache
+      return next();
+    } else if (/^(block|status)/.test(method)) {
+      // BLOCK CHECK: short cache
+      duration = oneBlock;
+    } else if (/^Query\/(Votes|TallyResult|Deposits)/.test(params)) {
+      // LIVE DATA: 5 minutes
+      duration = 60 * 5;
+    } else if (/^Query\/(Params|Pool)/.test(params)) {
+      // chain PARAMS: cache long-term (1 day)
+      duration = 60 * 60 * 24;
```

Comment:
I bumped Pool up to 1 day. This reflects total staking percentage I believe, which I think should be acceptable if it is a day old. LMK if this sounds off @jnaviask

Reply:
I may still need to tweak some keys. I think some things are getting invalidated when I don't want them to.

Reply:
I want to understand on-chain side little more, to understand how we can instead do invalidation, and not deal with different timeouts. 

Or, if we can control query in way that results can be cached for longer!

It can be a follow up if we think current timeouts are stable

If possible please do one sync session tomm.

Reply:
>I may still need to tweak some keys. I think some things are getting invalidated when I don't want them to.

resolved